pages/index.tsx

```diff
@@ -1,33 +1,25 @@
 import Head from 'next/head'
-import Layout, { siteTitle } from '../components/layout'
+import Layout from '../components/layout'
 import utilStyles from '../styles/utils.module.css'
 import { getSortedPostsData } from '../lib/posts'
 import Link from 'next/link'
 import Date from '../components/date'
+import { GetStaticProps } from 'next'
 
-export async function getStaticProps() {
-  const allPostsData = getSortedPostsData()
-  return {
-    props: {
-      allPostsData
-    }
-  }
-}
-
-export default function Home({ allPostsData }) {
+export default function Home({ allPostsData }: { allPostsData: { title: string; date: string; id: string; }[] }) {
   return (
     <Layout home>
       <section className={`${utilStyles.headingMd} ${utilStyles.padding1px}`}>
         <h2 className={utilStyles.headingLg}>Blog</h2>
         <ul className={utilStyles.list}>
-          {allPostsData.map(({ id, date, title }) => (
-            <li className={utilStyles.listItem} key={id}>
-              <Link href={`/posts/${id}`}>
-                <a>{title}</a>
+          {allPostsData.map((PostData) => (
```

Comment:
here, you are naming `PostData` the variable itself, which is confusing. We use upper CamelCase for classes, types, interfaces. And lower camelCase for variables, fields, data, etc. 
```suggestion
          {allPostsData.map((postData: PostData) => (
```
now after renaming it you should also rename the rest of places in this function where you were using `PostData` -> `postData`